### ISSUE TITLE
lipics-v2021: Problem with numberwithinsect in combination with thm-restate,cleveref, autoref,

### ISSUE THREAD

**Issue by java7mike** [dagstuhl-publishing/styles#15]
With options thm-restate,cleveref, autoref, there is no problem.

With options thm-restate,numberwithinsect,cleveref, autoref, I get a
pdfTeX warning (ext4) that Theorems 2.1 is using the same identifier
as Theorem 1.1, similarly 2.2 and 1.2. The file itself still uses
correct numbering, Theorems 1.1, 1.2, 2.1, 2.2, 3.1, and Theorem 1.2
is restated correctly. But hyperref links to 1.1 and 2.1 take me back
to 1.1, and hyperref links to both 1.2 and 2.2 take me back to 1.2.

MWE:
```
\documentclass[a4paper,UKenglish, numberwithinsect,thm-restate,cleveref,autoref]{lipics-v2021}

%\documentclass[a4paper,UKenglish,thm-restate,numberwithinsect,cleveref,autoref]{lipics-v2021}

%\documentclass[a4paper,UKenglish,thm-restate,cleveref, autoref]{lipics-v2021}

\title{Title}
\author{Author}{Affil}{}{}{}

\begin{document}
\maketitle

\section{One}\label{sec-one}

\begin{theorem}\label{thm-one}
Some text
\end{theorem}

\begin{restatable}{theorem}{restating}\label{thm-restate}
Some text for a restatable theorem
\end{restatable}

\newpage

\section{Two}\label{sec-two}

\begin{theorem}\label{thm-two-one}
Some text
\end{theorem}

\begin{theorem}\label{thm-two-two}
Some text
\end{theorem}

\newpage

\section{Three}\label{sec-three}

Can I link with hyperref correctly to
\cref{thm-one}?
What about \cref{thm-restate}?

What about \cref{thm-two-one}? and \cref{thm-two-two}?

  
\begin{theorem}\label{thm-three}
Some text
\end{theorem}

Can I restate a theorem?
\restating*
\end{document}
```

Resulting PDF: [mwe.pdf](https://github.com/dagstuhl-publishing/styles/files/6419125/mwe.pdf)

**Comment by java7mike** [comment]
At least in the MWE I was successful with the following adjustment:
Instead of the documentclass-option `numberwithinsec` I added the command `\numberwithin{theorem}{section}` to the LateX file (e.g. directly after the documentclass). (You may have to compile your document several times). 

Some more testing needs to be done before the style can be updated.

**Comment by java7mike** [comment]
@hanlin-ren Maybe you can test the above mentioned fix as well?

**Comment by hanlin-ren** [comment]
@java7mike Sorry but it didn't work for my MWE. After I removed `numberwithinsect` and added `\numberwithin{theorem}{section}`, the labels of the lemmas become two "Lemma 1"s instead of one "Lemma 1.1" and one "Lemma 2.1". When I click the hyperlink it still points to the first lemma.

**Comment by java7mike** [comment]
Thanks @hanlin-ren
You are absotely right, that `\numberwithin{theorem}{section}` alone doesn't solve the issue but it seems like the combination of `numberwithsect' and `\numberwithin{thereom}{section}` fixes the problem.

Does the following MWE work for you?

```
\documentclass[a4paper,cleveref, thm-restate, numberwithinsect, UKenglish]{lipics-v2021}


\numberwithin{theorem}{section}

\begin{document}

\section{sec 1}
\begin{lemma}
	sec 1
\end{lemma}

\newpage

\section{sec 2}
\begin{lemma}
	\label{lemma}
	sec 2
\end{lemma}
\cref{lemma}

\end{document}
```

**Comment by hanlin-ren** [comment]
@java7mike Sorry but it still doesn't work. When I click the link generated by `\cref` it points to "Lemma 1.1" instead of "Lemma 2.1".

**Comment by java7mike** [comment]
@hanlin-ren That's a pity. As it works for me, I assume that you have a slightly different LaTeX setup. Could you insert the command `\listfiles` in your document directly under the `\documentclass`? This will list all used packages including version information in the logfile. Could you then post the list of files here. 

Mine looks as following:
```
 *File List*
lipics-v2021.cls    2021/01/04 v3.1.0 LIPIcs articles
 article.cls    2020/04/10 v1.4m Standard LaTeX document class
   fleqn.clo    2016/12/29 v1.2b Standard LaTeX option (flush left equations)
  size10.clo    2020/04/10 v1.4m Standard LaTeX file (size option)
microtype.sty    2021/03/14 v2.8c Micro-typographical refinements (RS)
  keyval.sty    2014/10/28 v1.15 key=value parser (DPC)
microtype-pdftex.def    2021/03/14 v2.8c Definitions specific to pdftex (RS)
microtype.cfg    2021/03/14 v2.8c microtype main configuration file (RS)
inputenc.sty    2020/08/01 v1.3d Input encoding file
 lmodern.sty    2015/05/01 v1.6.1 Latin Modern Fonts
fontawesome5.sty    2020/03/24 v5.13.0 Font Awesome 5
   expl3.sty    2021-05-07 L3 programming layer (loader) 
l3backend-pdftex.def    2021-05-07 L3 backend support: PDF output (pdfTeX)
l3keys2e.sty    2021-05-07 LaTeX2e option processing using LaTeX3 keys
  xparse.sty    2021-01-09 L3 Experimental document command parser
xparse-2020-10-01.sty    
fontawesome5-generic-helper.sty    2020/03/24 v5.13.0 non-uTeX helper for fonta
wesome5
fontawesome5-mapping.def
 fontenc.sty
   t1lmr.fd    2015/05/01 v1.6.1 Font defs for Latin Modern
textcomp.sty    2020/02/02 v2.0n Standard LaTeX package
   eucal.sty    2009/06/22 v3.00 Euler Script fonts
 amssymb.sty    2013/01/14 v3.01 AMS font symbols
amsfonts.sty    2013/01/14 v3.01 Basic AMSFonts support
    soul.sty    2003/11/17 v2.4 letterspacing/underlining (mf)
   color.sty    1999/02/16
   color.cfg    2016/01/02 v1.6 sample color configuration
  pdftex.def    2020/10/05 v1.2a Graphics/color driver for pdftex
   babel.sty    2021/04/26 3.58 The Babel package
UKenglish.ldf    2017/06/06 v3.3r English support from the babel system
 amsmath.sty    2020/09/23 v2.17i AMS math features
 amstext.sty    2000/06/29 v2.01 AMS text
  amsgen.sty    1999/11/30 v2.0 generic functions
  amsbsy.sty    1999/11/29 v1.2d Bold Symbols
  amsopn.sty    2016/03/08 v2.02 operator names
enumerate.sty    2015/07/23 v3.00 enumerate extensions (DPC)
graphicx.sty    2020/09/09 v1.2b Enhanced LaTeX Graphics (DPC,SPQR)
graphics.sty    2020/08/30 v1.4c Standard LaTeX Graphics (DPC,SPQR)
    trig.sty    2016/01/03 v1.10 sin cos tan (DPC)
graphics.cfg    2016/06/04 v1.11 sample graphics configuration
   array.sty    2020/10/01 v2.5c Tabular extension package (FMi)
multirow.sty    2021/03/15 v2.8 Span multiple rows of a table
tabularx.sty    2020/01/15 v2.11c `tabularx' package (DPC)
threeparttable.sty    2003/06/13  v 3.0
listings.sty    2020/03/24 1.8d (Carsten Heinz)
 lstmisc.sty    2020/03/24 1.8d (Carsten Heinz)
listings.cfg    2020/03/24 1.8d listings configuration
  lineno.sty    2005/11/02 line numbers on paragraphs v4.41
totpages.sty    2005/09/19 v2.00 Totpages Package (muewi)
everyshi-ltx.sty    2020/08/17 v1.0a Emulation of the original everyshi package

with kernel methods
hyperref.sty    2021-02-27 v7.00k Hypertext links for LaTeX
 ltxcmds.sty    2020-05-10 v1.25 LaTeX kernel commands for general use (HO)
   iftex.sty    2020/03/06 v1.0d TeX engine tests
pdftexcmds.sty    2020-06-27 v0.33 Utility functions of pdfTeX for LuaTeX (HO)
infwarerr.sty    2019/12/03 v1.5 Providing info/warning/error messages (HO)
kvsetkeys.sty    2019/12/15 v1.18 Key value parser (HO)
kvdefinekeys.sty    2019-12-19 v1.6 Define keys (HO)
pdfescape.sty    2019/12/09 v1.15 Implements pdfTeX's escape features (HO)
 hycolor.sty    2020-01-27 v1.10 Color options for hyperref/bookmark (HO)
letltxmacro.sty    2019/12/03 v1.6 Let assignment for LaTeX macros (HO)
 auxhook.sty    2019-12-17 v1.6 Hooks for auxiliary files (HO)
kvoptions.sty    2020-10-07 v3.14 Key value format for package options (HO)
  pd1enc.def    2021-02-27 v7.00k Hyperref: PDFDocEncoding definition (HO)
hyperref-langpatches.def    2021-02-27 v7.00k Hyperref: patches for babel langu
ages
 intcalc.sty    2019/12/15 v1.3 Expandable calculations with integers (HO)
etexcmds.sty    2019/12/15 v1.7 Avoid name clashes with e-TeX commands (HO)
   puenc.def    2021-02-27 v7.00k Hyperref: PDF Unicode definition (HO)
     url.sty    2013/09/16  ver 3.4  Verb mode for urls, etc.
  bitset.sty    2019/12/09 v1.3 Handle bit-vector datatype (HO)
bigintcalc.sty    2019/12/15 v1.5 Expandable calculations on big integers (HO)
atbegshi-ltx.sty    2020/08/17 v1.0a Emulation of the original atbegshi package

with kernel methods
 hpdftex.def    2021-02-27 v7.00k Hyperref driver for pdfTeX
atveryend-ltx.sty    2020/08/19 v1.0a Emulation of the original atvery package
with kernel methods
rerunfilecheck.sty    2019/12/05 v1.9 Rerun checks for auxiliary files (HO)
uniquecounter.sty    2019/12/15 v1.4 Provide unlimited unique counter (HO)
 caption.sty    2020/10/26 v3.5g Customizing captions (AR)
caption3.sty    2020/10/21 v2.2e caption3 kernel (AR)
rotating.sty    2016/08/11 v2.16d rotated objects in LaTeX
  ifthen.sty    2014/09/29 v1.1c Standard LaTeX ifthen package (DPC)
subcaption.sty    2020/10/07 v1.3j Sub-captions (AR)
 xstring.sty    2019/02/06 v1.83 String manipulations (CT)
 comment.sty    
  amsthm.sty    2020/05/29 v2.20.6
thm-restate.sty    2020/08/01 v0.72
thmtools.sty    2020/08/01 v0.72
thm-patch.sty    2020/08/01 v0.72
parseargs.sty    2020/08/01 v0.72
  thm-kv.sty    2020/08/01 v0.72
thm-autoref.sty    2020/08/01 v0.72
aliasctr.sty    2020/08/01 v0.72
thm-listof.sty    2020/08/01 v0.72
thm-amsthm.sty    2020/08/01 v0.72
cleveref.sty    2018/03/27 v0.21.4 Intelligent cross-referencing
  mt-cmr.cfg    2013/05/19 v2.2 microtype config. file: Computer Modern Roman (
RS)
  xcolor.sty    2016/05/11 v2.12 LaTeX color extensions (UK)
   color.cfg    2016/01/02 v1.6 sample color configuration
supp-pdf.mkii
epstopdf-base.sty    2020-01-24 v2.11 Base part for package epstopdf
epstopdf-sys.cfg    2010/07/13 v1.3 Configuration of (r)epstopdf for TeX Live
 nameref.sty    2021-04-02 v2.47 Cross-referencing by name of section
refcount.sty    2019/12/15 v3.6 Data extraction from label references (HO)
gettitlestring.sty    2019/12/15 v1.6 Cleanup title references (HO)
     mwe.out
     mwe.out
  t1lmss.fd    2015/05/01 v1.6.1 Font defs for Latin Modern
  ot1lmr.fd    2015/05/01 v1.6.1 Font defs for Latin Modern
  omllmm.fd    2015/05/01 v1.6.1 Font defs for Latin Modern
 omslmsy.fd    2015/05/01 v1.6.1 Font defs for Latin Modern
 omxlmex.fd    2015/05/01 v1.6.1 Font defs for Latin Modern
    umsa.fd    2013/01/14 v3.01 AMS symbols A
  mt-msa.cfg    2006/02/04 v1.1 microtype config. file: AMS symbols (a) (RS)
    umsb.fd    2013/01/14 v3.01 AMS symbols B
  mt-msb.cfg    2005/06/01 v1.0 microtype config. file: AMS symbols (b) (RS)
 ***********
 ```

**Comment by hanlin-ren** [comment]
@java7mike The file list is as follows:
```
 *File List*
lipics-v2021.cls    2021/02/25 v3.1.1 LIPIcs articles
 article.cls    2019/12/20 v1.4l Standard LaTeX document class
   fleqn.clo    2016/12/29 v1.2b Standard LaTeX option (flush left equations)
  size10.clo    2019/12/20 v1.4l Standard LaTeX file (size option)
microtype.sty    2019/11/18 v2.7d Micro-typographical refinements (RS)
  keyval.sty    2014/10/28 v1.15 key=value parser (DPC)
microtype-pdftex.def    2019/11/18 v2.7d Definitions specific to pdftex (RS)
microtype.cfg    2019/11/18 v2.7d microtype main configuration file (RS)
inputenc.sty    2018/08/11 v1.3c Input encoding file
 lmodern.sty    2009/10/30 v1.6 Latin Modern Fonts
fontawesome5.sty    2020/01/17 v5.12.0.1 Font Awesome 5
   expl3.sty    2020-02-14 L3 programming layer (loader) 
l3backend-pdfmode.def    2020-02-03 L3 backend support: PDF mode
l3keys2e.sty    2020-02-14 LaTeX2e option processing using LaTeX3 keys
  xparse.sty    2020-02-14 L3 Experimental document command parser
fontawesome5-generic-helper.sty    2020/01/17 v5.12.0.1 non-uTeX helper for fon
tawesome5
fontawesome5-mapping.def
 fontenc.sty
   t1lmr.fd    2009/10/30 v1.6 Font defs for Latin Modern
textcomp.sty    2020/02/02 v2.0n Standard LaTeX package
   eucal.sty    2009/06/22 v3.00 Euler Script fonts
 amssymb.sty    2013/01/14 v3.01 AMS font symbols
amsfonts.sty    2013/01/14 v3.01 Basic AMSFonts support
    soul.sty    2003/11/17 v2.4 letterspacing/underlining (mf)
   color.sty    1999/02/16
   color.cfg    2016/01/02 v1.6 sample color configuration
  pdftex.def    2018/01/08 v1.0l Graphics/color driver for pdftex
   babel.sty    2020/02/14 3.40 The Babel package
UKenglish.ldf    2017/06/06 v3.3r English support from the babel system
 amsmath.sty    2020/01/20 v2.17e AMS math features
 amstext.sty    2000/06/29 v2.01 AMS text
  amsgen.sty    1999/11/30 v2.0 generic functions
  amsbsy.sty    1999/11/29 v1.2d Bold Symbols
  amsopn.sty    2016/03/08 v2.02 operator names
enumerate.sty    2015/07/23 v3.00 enumerate extensions (DPC)
graphicx.sty    2019/11/30 v1.2a Enhanced LaTeX Graphics (DPC,SPQR)
graphics.sty    2019/11/30 v1.4a Standard LaTeX Graphics (DPC,SPQR)
    trig.sty    2016/01/03 v1.10 sin cos tan (DPC)
graphics.cfg    2016/06/04 v1.11 sample graphics configuration
   array.sty    2019/08/31 v2.4l Tabular extension package (FMi)
multirow.sty    2019/05/31 v2.5 Span multiple rows of a table
tabularx.sty    2020/01/15 v2.11c `tabularx' package (DPC)
threeparttable.sty    2003/06/13  v 3.0
listings.sty    2019/09/10 1.8c (Carsten Heinz)
 lstmisc.sty    2019/09/10 1.8c (Carsten Heinz)
listings.cfg    2019/09/10 1.8c listings configuration
  lineno.sty    2005/11/02 line numbers on paragraphs v4.41
totpages.sty    2005/09/19 v2.00 Totpages Package (muewi)
everyshi.sty    2001/05/15 v3.00 EveryShipout Package (MS)
hyperref.sty    2020/01/14 v7.00d Hypertext links for LaTeX
 ltxcmds.sty    2019/12/15 v1.24 LaTeX kernel commands for general use (HO)
   iftex.sty    2019/11/07 v1.0c TeX engine tests
pdftexcmds.sty    2019/11/24 v0.31 Utility functions of pdfTeX for LuaTeX (HO)
infwarerr.sty    2019/12/03 v1.5 Providing info/warning/error messages (HO)
kvsetkeys.sty    2019/12/15 v1.18 Key value parser (HO)
kvdefinekeys.sty    2019-12-19 v1.6 Define keys (HO)
pdfescape.sty    2019/12/09 v1.15 Implements pdfTeX's escape features (HO)
 hycolor.sty    2020-01-27 v1.10 Color options for hyperref/bookmark (HO)
letltxmacro.sty    2019/12/03 v1.6 Let assignment for LaTeX macros (HO)
 auxhook.sty    2019-12-17 v1.6 Hooks for auxiliary files (HO)
kvoptions.sty    2019/11/29 v3.13 Key value format for package options (HO)
  pd1enc.def    2020/01/14 v7.00d Hyperref: PDFDocEncoding definition (HO)
 intcalc.sty    2019/12/15 v1.3 Expandable calculations with integers (HO)
etexcmds.sty    2019/12/15 v1.7 Avoid name clashes with e-TeX commands (HO)
   puenc.def    2020/01/14 v7.00d Hyperref: PDF Unicode definition (HO)
     url.sty    2013/09/16  ver 3.4  Verb mode for urls, etc.
  bitset.sty    2019/12/09 v1.3 Handle bit-vector datatype (HO)
bigintcalc.sty    2019/12/15 v1.5 Expandable calculations on big integers (HO)
atbegshi.sty    2019/12/05 v1.19 At begin shipout hook (HO)
 hpdftex.def    2020/01/14 v7.00d Hyperref driver for pdfTeX
atveryend.sty    2019-12-11 v1.11 Hooks at the very end of document (HO)
rerunfilecheck.sty    2019/12/05 v1.9 Rerun checks for auxiliary files (HO)
uniquecounter.sty    2019/12/15 v1.4 Provide unlimited unique counter (HO)
 caption.sty    2020/01/03 v3.4h Customizing captions (AR)
caption3.sty    2020/01/03 v1.8h caption3 kernel (AR)
rotating.sty    2016/08/11 v2.16d rotated objects in LaTeX
  ifthen.sty    2014/09/29 v1.1c Standard LaTeX ifthen package (DPC)
subcaption.sty    2020/01/22 v1.3d Sub-captions (AR)
 xstring.sty    2019/02/06 v1.83 String manipulations (CT)
 comment.sty    
  amsthm.sty    2017/10/31 v2.20.4
thm-restate.sty    2019/12/22 v68
thmtools.sty    2019/12/22 v68
thm-patch.sty    2019/12/22 v68
parseargs.sty    2019/12/22 v68
  thm-kv.sty    2019/12/22 v68
thm-autoref.sty    2019/12/22 v68
aliasctr.sty    2019/12/22 v68
thm-listof.sty    2019/12/22 v68
thm-amsthm.sty    2019/12/22 v68
cleveref.sty    2018/03/27 v0.21.4 Intelligent cross-referencing
  mt-cmr.cfg    2013/05/19 v2.2 microtype config. file: Computer Modern Roman (
RS)
supp-pdf.mkii
epstopdf-base.sty    2020-01-24 v2.11 Base part for package epstopdf
  grfext.sty    2019/12/03 v1.3 Manage graphics extensions (HO)
epstopdf-sys.cfg    2010/07/13 v1.3 Configuration of (r)epstopdf for TeX Live
  xcolor.sty    2016/05/11 v2.12 LaTeX color extensions (UK)
   color.cfg    2016/01/02 v1.6 sample color configuration
 nameref.sty    2019/09/16 v2.46 Cross-referencing by name of section
refcount.sty    2019/12/15 v3.6 Data extraction from label references (HO)
gettitlestring.sty    2019/12/15 v1.6 Cleanup title references (HO)
 article.out
 article.out
  t1lmss.fd    2009/10/30 v1.6 Font defs for Latin Modern
  ot1lmr.fd    2009/10/30 v1.6 Font defs for Latin Modern
  omllmm.fd    2009/10/30 v1.6 Font defs for Latin Modern
 omslmsy.fd    2009/10/30 v1.6 Font defs for Latin Modern
 omxlmex.fd    2009/10/30 v1.6 Font defs for Latin Modern
    umsa.fd    2013/01/14 v3.01 AMS symbols A
  mt-msa.cfg    2006/02/04 v1.1 microtype config. file: AMS symbols (a) (RS)
    umsb.fd    2013/01/14 v3.01 AMS symbols B
  mt-msb.cfg    2005/06/01 v1.0 microtype config. file: AMS symbols (b) (RS)
 ***********
```

**Comment by java7mike** [comment]
At least in our LaTeX environments, the fix provided in #19 seems to resolve the issues. (We have several environments for testing based on texlive2019, texlive2020, and texlive2021).


MWE:
```
\documentclass[a4paper,UKenglish, numberwithinsect,thm-restate,cleveref,autoref]{lipics-v2021}

%\documentclass[a4paper,UKenglish,thm-restate,numberwithinsect,cleveref,autoref]{lipics-v2021}

%\documentclass[a4paper,UKenglish,thm-restate,cleveref, autoref]{lipics-v2021}

\title{Title}
\author{Author}{Affil}{}{}{}

\begin{document}
\maketitle

\section{One}\label{sec-one}

\begin{theorem}\label{thm-one}
Some text
\end{theorem}

\begin{restatable}{theorem}{restating}\label{thm-restate}
Some text for a restatable theorem
\end{restatable}

\newpage

\section{Two}\label{sec-two}

\begin{theorem}\label{thm-two-one}
Some text
\end{theorem}

\begin{theorem}\label{thm-two-two}
Some text
\end{theorem}

\newpage

\section{Three}\label{sec-three}

Can I link with hyperref correctly to
\cref{thm-one}?
What about \cref{thm-restate}?

What about \cref{thm-two-one}? and \cref{thm-two-two}?

  
\begin{theorem}\label{thm-three}
Some text
\end{theorem}

Can I restate a theorem?
\restating*
\end{document}
```

Resulting PDF: [mwe-v4.pdf](https://github.com/dagstuhl-publishing/styles/files/6559201/mwe-v4.pdf)

**Comment by java7mike** [comment]
@hanlin-ren Unfortunately, I wasn't able to reproduce the issue reported by you. Even with a nearly identical LaTeX environment, the fix seems to work for me correctly.

MWE:
```
\documentclass[a4paper,cleveref, thm-restate, numberwithinsect, UKenglish]{lipics-v2021}

\listfiles

\begin{document}

\section{sec 1}
\begin{lemma}
	sec 1
\end{lemma}

\newpage

\section{sec 2}
\begin{lemma}
	\label{lemma}
	sec 2
\end{lemma}
\cref{lemma}

\end{document}
```

Resulting PDF: [mwe-v2.pdf](https://github.com/dagstuhl-publishing/styles/files/6559221/mwe-v2.pdf)

**Comment by hanlin-ren** [comment]
@java7mike the link in the PDF above works correctly.

I changed my local lipics-v2021.cls according to your pull request #19. Now everything works correctly on my computer. Thank you!

**Comment by a3nm** [comment]
Hi, I was bitten by this bug too and the discussion here was very helpful. Sorry though, but I'm confused: it looks like the current master version of lipics-v2021.cls, [here](https://github.com/dagstuhl-publishing/styles/commits/master/LIPIcs/authors/lipics-v2021.cls), does *not* incorporate the fix from #19 that solves this problem ([here](https://github.com/dagstuhl-publishing/styles/commits/6757c80b29f95db47ef431b880b55ad8be8c04d6/LIPIcs/authors/lipics-v2021.cls)), even though the master version has an (apparently unrelated) commit with a later date (?!).

I'm probably confused by your tagging scheme, but are there plans to ship the fix to this bug in the master version of lipics-v2021.cls?

Thanks!
